### PR TITLE
[PAN-2362] permissioning cli smart contract

### PR DESCRIPTION
**Pull request by macfarla**
## PR description
added CLI options for smart contract node permissioning
--permissions-nodes-contract-address
--permissions-nodes-contract-enabled

## Fixed Issue(s)
fixes #PAN-2362

### CODE REVIEW

File: pantheon/src/test/java/tech/pegasys/pantheon/cli/PantheonCommandTest.java [hunk 1]
```diff
@@ -307,6 +308,67 @@ public void overrideDefaultValuesIfKeyIsPresentInConfigFile() throws IOException
...
+
+    verifyZeroInteractions(mockRunnerBuilder);
+
+    assertThat(commandErrorOutput.toString())
+        .startsWith("Missing required parameter for option '--permissions-nodes-contract-address'");
+    assertThat(commandOutput.toString()).isEmpty();
+  }
+
+  @Test
+  public void permissionsEnabledWithoutContractAddressMustNotError() {
```
Comment (lucassaldanha) [inline pantheon/src/test/java/tech/pegasys/pantheon/cli/PantheonCommandTest.java]: I believe we want the CLI to fail when the user enables smart contract based permissioning but doesn't specify the contract address. We can't do smart contract permissioning without the address.

File: pantheon/src/main/java/tech/pegasys/pantheon/cli/PantheonCommand.java [hunk 2]
```diff
@@ -791,25 +796,74 @@ MetricsConfiguration metricsConfiguration() {
...
+    }
+    if (contractPermissionsEnabled()) {
+      if (permissionsNodesContractAddress != null) {
+        final SmartContractPermissioningConfiguration smartContractPermissioningConfiguration =
+            PermissioningConfigurationBuilder.smartContractPermissioningConfiguration(
+                permissionsNodesContractAddress, permissionsNodesContractEnabled);
+        smartContractPermissioningConfigurationOptional =
+            Optional.of(smartContractPermissioningConfiguration);
+      } else {
+        logger.warn("No contract address specified. Cannot enable contract permissioning.");
```
Comment (lucassaldanha) [inline pantheon/src/main/java/tech/pegasys/pantheon/cli/PantheonCommand.java]: Should this be just a warning? If there is no address, there is no point in enabling smart contract permissioning. And it feels like it is a really bad thing if the user doesn't see the warning and thinks that permissioning has been set.

File: pantheon/src/test/java/tech/pegasys/pantheon/cli/PantheonCommandTest.java (hunk 1 above)
Comment (macfarla) [reply pantheon/src/test/java/tech/pegasys/pantheon/cli/PantheonCommandTest.java]: agreed

Comment (macfarla) [reply pantheon/src/test/java/tech/pegasys/pantheon/cli/PantheonCommandTest.java]: this test now errors

File: pantheon/src/main/java/tech/pegasys/pantheon/cli/PantheonCommand.java (hunk 2 above)
Comment (macfarla) [reply pantheon/src/main/java/tech/pegasys/pantheon/cli/PantheonCommand.java]: agree. added ParameterException in this case.

File: pantheon/src/main/java/tech/pegasys/pantheon/cli/PantheonCommand.java [hunk 3]
```diff
@@ -603,13 +615,6 @@ public void run() {
               + "or specify the beneficiary of mining (via --miner-coinbase <Address>)");
     }
 
-    if (permissionsConfigFile() != null) {
-      if (!permissionsAccountsEnabled && !permissionsNodesEnabled) {
-        logger.warn(
-            "Permissions config file set {} but no permissions enabled", permissionsConfigFile());
-      }
-    }
-
```
Comment (macfarla) [inline pantheon/src/main/java/tech/pegasys/pantheon/cli/PantheonCommand.java]: moved to permissioningConfiguration() method

**Pull request merged**